bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultModelController.java

```diff
@@ -253,28 +255,26 @@ public void getValidationConstraints(final Context ctx, final String modeluri) {
 
    @Override
    public void undo(final Context ctx, final String modeluri) {
-      Optional<CCommandExecutionResult> undoExecution = modelRepository.undo(modeluri);
-      if (undoExecution.isPresent()) {
-         // TODO: Split v1 and v2 results
-         // In V2, send a jsonPatch response
-         success(ctx, "Successful undo.");
-         sessionController.commandExecuted(modeluri, undoExecution.get());
-         return;
-      }
-      accepted(ctx, "Cannot undo");
+      withModel(ctx, modeluri, root -> {
+         modelRepository.undo(modeluri).ifPresentOrElse(undoExecution -> {
+            success(ctx, "Successful undo.");
```

Comment:
Should we return the patch here as well, similar to what's done for executePatchCommand?

Reply:
Yes, of course. 🤦